File: apps/frontend/mobile/src/hooks/team/teamView.ts
import type {
  MobileTeamDetail,
  MobileTeamSummary,
  MobileTeamUpdated,
} from '@/src/api/generated/mobile-gateway/models';
import type {
  EnrichedTeamDTO,
  TeamSummaryDTO,
  TeamUpdateResult,
} from '@/src/types/Team';
import {
  toCatalogDivisionView,
  toFormatView,
  toGenderView,
} from '@/src/hooks/catalog/catalogView';
import { toRankedPoolView } from '@/src/hooks/pool/poolView';

export function toEnrichedTeamView(
  response: MobileTeamDetail,
): EnrichedTeamDTO {
  return {
    id: response.id,
    clubId: response.clubId,
    name: response.name,
    shortName: response.shortName,
    rawName: response.rawName,
    format: toFormatView(response.format),
    gender: toGenderView(response.gender),
    season: response.season,
    followersCount: response.followersCount,
    division: toCatalogDivisionView(response.division),
    logoUrl: response.logoUrl,
    pools: response.pools.map(toRankedPoolView),
  };
}

export function toTeamSummaryView(response: MobileTeamSummary): TeamSummaryDTO {
  return {
    id: response.id,
    name: response.name,
    shortName: response.shortName,
    season: response.season,
    gender: toGenderView(response.gender),
    format: toFormatView(response.format),
    logoUrl: response.logoUrl,
    division: response.division
      ? toCatalogDivisionView(response.division)
      : null,
  };
}

export function toTeamUpdateResult(
  response: MobileTeamUpdated,
): TeamUpdateResult {
  return {
    id: response.id,
    name: response.name,
    shortName: response.shortName,
    logoUrl: response.logoUrl,
  };
}

File: apps/backend/mobile-gateway/src/main/java/com/blockout/mobilegateway/team/application/views/TeamView.java
package com.blockout.mobilegateway.team.application.views;

import com.blockout.mobilegateway.club.application.views.ClubView;
import com.blockout.mobilegateway.config.application.views.DivisionView;
import com.blockout.mobilegateway.pool.application.views.PoolView;
import com.blockout.mobilegateway.shared.application.models.Format;
import com.blockout.mobilegateway.shared.application.models.Gender;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TeamView {
    private Long id;

    private String name;

    private String clubId;

    private String shortName;

    private String rawName;

    private Format format;

    private Gender gender;

    private String season;

    private Long followersCount;

    private String logoUrl;

    private ClubView club;

    private DivisionView division;

    private List<PoolView> pools;
}
